fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/ambassador/simulation/perception/util/VehicleIndexProviderTypeAdapterFactory.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2022 Fraunhofer FOKUS and others. All rights reserved.
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contact: mosaic@fokus.fraunhofer.de
+ */
+
+package org.eclipse.mosaic.fed.application.ambassador.simulation.perception.util;
+
+import org.eclipse.mosaic.fed.application.ambassador.simulation.perception.index.providers.VehicleIndex;
+import org.eclipse.mosaic.lib.gson.AbstractTypeAdapterFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonParseException;
+import com.google.gson.TypeAdapter;
+import com.google.gson.TypeAdapterFactory;
+import com.google.gson.reflect.TypeToken;
+
+public class VehicleIndexProviderTypeAdapterFactory implements TypeAdapterFactory {
+    public static class VehicleIndexProviderTypeAdapter extends AbstractTypeAdapterFactory<VehicleIndex> {
+
+        private VehicleIndexProviderTypeAdapter(TypeAdapterFactory parentFactory, Gson gson) {
+            super(parentFactory, gson);
+        }
+
+        @Override
+        protected Class<?> fromTypeName(String type) {
+            try {
+                return Class.forName(VehicleIndex.class.getPackage().getName() + "." + type);
```

Comment:
What happens if you type "VehicleIndex" or "TrafficLightMap" into the config? This would not fail in this line, but probably later, and I just wonder what error message would be printed then?

Reply:
Luckily, our JSON-validation stops this beforehand.
Errors would look like this:
```
...
Caused by: java.lang.RuntimeException: java.lang.InstantiationException: The CApplicationAmbassador config is not valid: Exactly one of the following sets of problems must be resolved.
1) [4,37][/perceptionConfiguration/vehicleIndex/type] The value must be one of ["VehicleMap"].
2) [4,37][/perceptionConfiguration/vehicleIndex/type] The value must be one of ["VehicleTree"].
3) [4,37][/perceptionConfiguration/vehicleIndex/type] The value must be one of ["VehicleGrid"].
4) [4,37][/perceptionConfiguration/vehicleIndex/type] The value must be one of ["SumoIndex"].
```